editor/editor_node.cpp

```diff
@@ -2458,6 +2458,10 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 
 			settings_config_dialog->popup_edit_settings();
 		} break;
+		case SETTINGS_USER_DATA_FOLDER: {
+
+			OS::get_singleton()->shell_open(String("file://") +	ProjectSettings::get_singleton()->globalize_path("user://"));
```

Comment:
There's actually `OS::get_singleton()->get_user_data_dir()` to get this global path.